src/scrape_up/gitlab/users.py

```diff
@@ -0,0 +1,97 @@
+import requests
+from bs4 import BeautifulSoup
+
+class GitLab:
+    def __init__(self, username):
+        self.username = username
+
+    def get_name(self):
+    # URL for the GitLab user's profile page
+    url = f"https://gitlab.com/{self.username}"
+    
+    # Send a GET request to the profile page
+    response = requests.get(url)
+    
+    # Parse the HTML content using BeautifulSoup
+    soup = BeautifulSoup(response.text, "html.parser")
+    
+    # Extract profile name
+    name = soup.find("h1", class_="user-title").text.strip()
+    
+    return name
+
+
+def get_bio(self):
+    # URL for the GitLab user's profile page
+    url = f"https://gitlab.com/{self.username}"
+    
+    # Send a GET request to the profile page
+    response = requests.get(url)
+    
+    # Parse the HTML content using BeautifulSoup
+    soup = BeautifulSoup(response.text, "html.parser")
+    
+    # Extract profile bio
+    bio = soup.find("div", class_="user-info").text.strip()
+    
+    return bio
+
+
+def get_avatar_url(self):
+    # URL for the GitLab user's profile page
+    url = f"https://gitlab.com/{self.username}"
+    
+    # Send a GET request to the profile page
+    response = requests.get(url)
+    
+    # Parse the HTML content using BeautifulSoup
+    soup = BeautifulSoup(response.text, "html.parser")
+    
+    # Extract avatar URL
+    avatar_url = soup.find("img", class_="avatar s40")["src"]
+    
+    return avatar_url
+
+    def get_repositories(self):
+        # URL for the GitLab user's repositories page
+        url = f"https://gitlab.com/{self.username}?tab=repositories"
+        
+        # Send a GET request to the repositories page
+        response = requests.get(url)
+        
+        # Parse the HTML content using BeautifulSoup
+        soup = BeautifulSoup(response.text, "html.parser")
+        
+        # Extract repository information
+        repositories = []
+        repo_elements = soup.find_all("li", class_="project-row")
+        for repo_element in repo_elements:
+            name = repo_element.find("h3", class_="project-title").text.strip()
+            description = repo_element.find("p", class_="project-description").text.strip()
+            repositories.append({
+                "name": name,
+                "description": description
+            })
+        
+        # Return the list of repositories
+        return repositories
```

Comment:
Same as others, the return method is not implemented as per the project structure, also no try-catch is used as well.
Please look into it